scripts/ci-e2e.sh

```diff
@@ -61,6 +61,7 @@ export E2E_CONF_FILE="${REPO_ROOT}/test/e2e/config/docker.yaml"
 export ARTIFACTS="${ARTIFACTS:-${REPO_ROOT}/_artifacts}"
 export SKIP_RESOURCE_CLEANUP=false
 export USE_EXISTING_CLUSTER=false
+export KIND_EXPERIMENTAL_DOCKER_NETWORK="bridge"
```

Comment:
Wondering if we can get CAPD to create machines in the `kind` network by default (with eventually support for `KIND_EXPERIMENTAL_DOCKER_NETWORK` for handling exceptions).
This will simplify both our dev workflow and our quick start experience.

Reply:
Do you mind opening an issue and assigning it to me so I can look in to it?

Reply:
I'm not sure this deserves another issue, because most probably it will be enough to set the network = "kind" in this bit of code https://github.com/kubernetes-sigs/cluster-api/blob/f631989d3f7430b5825644a5461e116f48ed2e28/test/infrastructure/docker/docker/kind_manager.go#L99-L155

However, if you prefer I can investigate the topic and open a separated issue, but in this case we have to document the usage of the variable first, and the remove

Reply:
Opened https://github.com/kubernetes-sigs/cluster-api/issues/3832.